### PR TITLE
Resolve #25948, fixing the custom attribute filter when using configurable products

### LINKED ISSUE #25948: Custom Attributes for Configurable Products in Layered Navigation lead to errors in Category View
### Preconditions (*)

1. Installed Magento Version is 2.3.2
2. Indexers are configured to run on schedule

### Steps to reproduce (*)

1. Create a new custom attribute for products.
2. Create a new configurable product.
3. Assign the attribute to the main product and all it's children.
4. Make sure the attribute has the "Use in Layered Navigation" attribute set to "Filterable (with results)"
5. Run reindex manually bin/magento index:reindex

### Expected result (*)

1. The layered navigation shows the new property.
2. When I click the property to filter the catalog items the list is filtered automatically.

### Actual result (*)

1. The following error is thrown:

```
There has been an error processing your request
Item (Magento\Catalog\Model\Product\Interceptor) with the same ID "4046" already exists.
```

### PR DESCRIPTION
**Pull request by zingix**
### Description (*)
This pull request fixes the error described in #25948 when custom attributes and layered navigation is used along with configurable products.

### Fixed Issues (if relevant)

1. magento/magento2#25948: Custom Attributes for Configurable Products in Layered Navigation lead to errors in Category View

### Manual testing scenarios (*)

1. See related issue. Use the same procedure.
2. You can also print the generated SQL and try it yourself. You will see that the inner join for the custom attribute will multiply the product catalog by the number of available configurable products when the attribute is set.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by zingix** [comment]
Hi @akaplya,
Please see my commits above. They get rejected but I guess it's a problem with the tests. I am not sure but we discovered in production that attributes are not mapped correctly. Can you have a look at it?

File: app/code/Magento/Catalog/Model/ResourceModel/Layer/Filter/Attribute.php [hunk 1]
```diff
@@ -41,6 +41,11 @@ public function applyFilterToCollection(\Magento\Catalog\Model\Layer\Filter\Filt
         $tableAlias = $attribute->getAttributeCode() . '_idx';
         $conditions = [
             "{$tableAlias}.entity_id = e.entity_id",
+            // exclude children of configurable products
+            $connection->quoteInto(
+                "(e.type_id <> ? OR {$tableAlias}.entity_id = {$tableAlias}.source_id)",
+                'configurable'
```
Comment (sidolov) [inline app/code/Magento/Catalog/Model/ResourceModel/Layer/Filter/Attribute.php]: Please, use `\Magento\ConfigurableProduct\Model\Product\Type\Configurable::TYPE_CODE` instead.

File: app/code/Magento/Catalog/Model/ResourceModel/Layer/Filter/Attribute.php [hunk 2]
```diff
@@ -77,6 +82,11 @@ public function getCount(\Magento\Catalog\Model\Layer\Filter\FilterInterface $fi
         $tableAlias = sprintf('%s_idx', $attribute->getAttributeCode());
         $conditions = [
             "{$tableAlias}.entity_id = e.entity_id",
+            // exclude children of configurable products
+            $connection->quoteInto(
+                "(e.type_id <> ? OR {$tableAlias}.entity_id = {$tableAlias}.source_id)",
+                'configurable'
```
Comment (sidolov) [inline app/code/Magento/Catalog/Model/ResourceModel/Layer/Filter/Attribute.php]: Please, use `\Magento\ConfigurableProduct\Model\Product\Type\Configurable::TYPE_CODE` instead.

**Review by sidolov — CHANGES_REQUESTED** [review]
@zingix please, cover changes with automated test

**Comment by engcom-Charlie** [comment]
HI @zingix, could you please cover changes by automated test and look through requested changes?

**Comment by sidolov** [comment]
@zingix please, cover changes with automated test

**Comment by engcom-Charlie** [comment]
Hi @zingix, I will close this PR because the issue that was to be solved by this PR has been closed.
Thank you for your collaboration.

**Pull request closed without merging**